js/main.js

```diff
@@ -30,13 +31,27 @@ function handleSubmitForm(event) {
     notes: $entryForm.notes.value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId++;
-  data.entries.unshift(formObject);
+  if (data.editing !== null) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.entries[i] = {
+          title: $entryForm.title.value,
+          url: $entryForm.url.value,
+          notes: $entryForm.notes.value,
+          entryId: data.editing.entryId
+        };
+      }
+    }
+  } else {
+    data.nextEntryId++;
+    data.entries.unshift(formObject);
+  }
+  $newEntriesObject.innerHTML = '';
+  DOMObjectLoaded();
```

Comment:
If you are going to use the function like this, it needs a different name, something like `generateEntryList`